src/core/unittest/RecvBufferTest.cpp

```diff
@@ -0,0 +1,380 @@
+/*++
+
+    Copyright (c) Microsoft Corporation.
+    Licensed under the MIT License.
+
+Abstract:
+
+    Unit test for the QUIC_RECV_BUFFER interface.
+
+--*/
+
+#include "main.h"
+#ifdef QUIC_CLOG
+#include "RecvBufferTest.cpp.clog.h"
+#endif
+
+#define DEF_TEST_BUFFER_LENGTH 64
+#define LARGE_TEST_BUFFER_LENGTH 1024
+
+struct RecvBuffer {
+    QUIC_RECV_BUFFER RecvBuf {0};
+    uint8_t* PreallocChunk {nullptr};
+    ~RecvBuffer() {
+        QuicRecvBufferUninitialize(&RecvBuf);
+        if (PreallocChunk) {
+            CXPLAT_FREE(PreallocChunk, QUIC_POOL_TEST);
+        }
+    }
+    QUIC_STATUS Initialize(
+        _In_ bool CopyOnDrain = true,
+        _In_ bool PreallocatedChunk = false,
+        _In_ uint32_t AllocBufferLength = DEF_TEST_BUFFER_LENGTH,
+        _In_ uint32_t VirtualBufferLength = DEF_TEST_BUFFER_LENGTH
+        ) {
+        if (PreallocatedChunk) {
+            PreallocChunk =
+                (uint8_t*)CXPLAT_ALLOC_NONPAGED(
+                    AllocBufferLength,
+                    QUIC_POOL_TEST);
+        }
+        return QuicRecvBufferInitialize(&RecvBuf, AllocBufferLength, VirtualBufferLength, CopyOnDrain ? TRUE : FALSE, PreallocChunk);
+    }
+    uint64_t GetTotalLength() {
+        return QuicRecvBufferGetTotalLength(&RecvBuf);
+    }
+    void SetVirtualBufferLength(uint32_t Length) {
+        QuicRecvBufferSetVirtualBufferLength(&RecvBuf, Length);
+    }
+    bool HasUnreadData() {
+        return QuicRecvBufferHasUnreadData(&RecvBuf) != FALSE;
```

Comment:
nit: no need to have "!= FALSE"?

Reply:
It's converting between `BOOLEAN` and `bool`.